app/View/Components/Community/UserCard.php

```diff
@@ -0,0 +1,163 @@
+        $cardRoleData = $this->buildCardRoleData($username, (int) $rawUserData['Permissions']);
+
+        return array_merge($cardBioData, $cardRankData, $cardRoleData);
+    }
+
+    private function buildCardBioData(array $rawUserData): array
+    {
+        $username = $rawUserData['User'] ?? "";
+        $motto = $rawUserData['Motto'] ?? null;
+        $avatarUrl = $rawUserData['avatarUrl'] ?? null;
+        $hardcorePoints = $rawUserData['RAPoints'] ?? 0;
+        $softcorePoints = $rawUserData['RASoftcorePoints'] ?? 0;
+        $retroPoints = $rawUserData['TrueRAPoints'] ?? 0;
+        $isUntracked = $rawUserData['Untracked'] ? true : false;
+        $permissions = $rawUserData['Permissions'] ?? Permissions::Unregistered;
+        $memberSince = $rawUserData['Created'] ?? Carbon::now();
+        $lastActivity = $rawUserData['LastLogin'] ? Carbon::parse($rawUserData['LastLogin'])->diffForHumans() : null;
+
+        return compact(
+            'username',
+            'motto',
+            'avatarUrl',
+            'hardcorePoints',
+            'softcorePoints',
+            'retroPoints',
+            'isUntracked',
+            'permissions',
+            'memberSince',
+            'lastActivity'
+        );
+    }
+
+    private function buildCardRankData(string $username, int $hardcorePoints, int $softcorePoints, bool $isUntracked): array
+    {
+        $siteRank = 0;
+        $totalRankedUsersCount = 0;
+        $rankType = RankType::Hardcore;
+        $rankLabel = 'Site Rank:';
+        $rankPctLabel = '';
+        $rankMinPoints = Rank::MIN_POINTS;
+
+        if ($isUntracked) {
+            $siteRank = 'Untracked';
+            $rankType = 'Untracked';
+        } elseif ($hardcorePoints >= $softcorePoints) {
+            $rankType = RankType::Hardcore;
+            $siteRank = $hardcorePoints < Rank::MIN_POINTS ? 0 : getUserRank($username, $rankType);
+        } elseif ($softcorePoints > 0) {
+            $rankType = RankType::Softcore;
+            $siteRank = $softcorePoints < Rank::MIN_POINTS ? 0 : getUserRank($username, $rankType);
+            $rankLabel = 'Softcore Rank:';
+        }
+
+        if ($rankType !== 'Untracked') {
+            $totalRankedUsersCount = countRankedUsers($rankType);
+
+            // Don't divide by zero.
+            if ($totalRankedUsersCount === 0) {
+                $totalRankedUsersCount = 1;
+            }
+
+            $rankPct = sprintf("%1.2f", ($siteRank / $totalRankedUsersCount) * 100.0);
+            $rankPctLabel = $siteRank > 100 ? "(Top $rankPct%)" : "";
+        }
+
+        return compact(
+            'siteRank',
+            'totalRankedUsersCount',
+            'rankType',
+            'rankLabel',
+            'rankPctLabel',
+            'rankMinPoints',
+        );
+    }
+
+    private function buildCardRoleData(string $username, int $permissions): array
+    {
+        $canShowUserRole = $permissions >= Permissions::JuniorDeveloper;
```

Comment:
Not sure if we care, but Banned and Spam accounts are no longer flagged as such on the card.

Reply:
I know we briefly talked about it in the forum refactor, and the decision was to not highlight the ban status in the forum entry user panel. The user-card does currently show the banned status, so I'm just noting that this is a change in behavior.

I think it's probably fine to not include it here.

There's still a bit of an oddity that clicking the user name will go to a Not Found page, but I don't think there's much of a demand for going to a user profile from a thread post.